frontend/dbg.lua

```diff
@@ -86,6 +86,13 @@ function Dbg:v(...)
     end
 end
 
+--- Conditional logging with a stable ref.
+function Dbg.log(...)
+    if Dbg.is_on then
+        return LvDEBUG(...)
+    end
+end
```

Comment:
Would it make sense to switch references around?

Dbg.log = function() end

and

Dbg.log = Dbg.forRealLvDEBUG

To avoid the extra condition.

Reply:
The condition is kind of the point, though.

The intended use-case is for funky stuff like what framebuffer does, where the class has a `debug` function that is NOP, that is replaced during instantiation by a reference to `logger.dbg` fed by the constructor.

The issue is that Logger mutates its methods, so if debugging is disabled, that's NOP, and that stale reference will stay pinned to NOP forever ;).

If you feed it that instead, it works as intended: logging follows the actual debugging state.

If it was instead declared as `Dbg.log = LvDEBUG` (or wrapped in a useless closure as `function Dbg.log = function(...) return LvDEBUG(...) end`, it would log *always*, even if debug is disabled.

(For context, `local LvDEBUG = logger.LvDEBUG`, and that's a Logger method that is *never* mutated, and always logs in under the "dbg" context).

Reply:
(Unless I completely missed your point, that how I understood the above ;)).

Reply:
My point was why only mutate on instantiation if this is desired functionality?

Reply:
Dbg:turnOn() and turnOff() already do this for guard() and dassert(). Is it so bad to mutate if the alternative is to always branch?

Reply:
But `guard` & `dassert` are always called via `dbg.`, we never store a reference to them in an assignment and then use *that* stale reference to call them, which is the whole difference. So we effectively always look up the current, live ref.

To be clear: I *like* the current behavior, despite its mild inconvenience of requiring a restart because of framebuffer's pinned ref.

I just wanted to provide an alternative if we ever needed something like that in the future, at the cost of a closure & a branch ;).

Reply:
Which leads me to a possibly much simpler solution: pass a ref to logger, instead of logger.dbg, and then just call logger.dbg(), that'll look up the current ref.

Reply:
> Which leads me to a possibly much simpler solution: pass a ref to logger, instead of logger.dbg, and then just call self.logger.dbg(), that'll look up the current ref.

Eh, that would probably make the testsuite cranky, unless we implement a stub logger class with all the right methods :/.

Reply:
> To be clear: I _like_ the current behavior, despite its mild inconvenience of requiring a restart because of framebuffer's pinned ref.

I'm not the one complaining about a restart; you're the one adding this branch here, lol